x-pack/plugins/apm/public/components/shared/Links/MachineLearningLinks/MLJobLink.tsx

```diff
@@ -8,22 +8,33 @@ import React from 'react';
 import { getMlJobId } from '../../../../../common/ml_job_constants';
 import { MLLink } from './MLLink';
 
-interface Props {
+interface PropsServiceName {
   serviceName: string;
   transactionType?: string;
 }
+interface PropsJobId {
+  jobId: string;
+}
+
+type Props = (PropsServiceName | PropsJobId) & {
+  external?: boolean;
+};
 
-export const MLJobLink: React.FC<Props> = ({
-  serviceName,
-  transactionType,
-  children
-}) => {
-  const jobId = getMlJobId(serviceName, transactionType);
+export const MLJobLink: React.FC<Props> = props => {
```

Comment:
Are there any advantages to having a union over making some of the props optional?

```js
interface Props {
  serviceName: string;
  transactionType?: string;
  jobId?: string;
  external?: boolean;
}

export const MLJobLink: React.FC<Props> = props => {
  const jobId = props.jobId
    ? props.jobId
    : getMlJobId(props.serviceName, props.transactionType);
```